src/util.js

```diff
@@ -168,3 +168,26 @@ function ComparativeDate(referenceDate, date, options) {
     // TODO isPrev, isNext
   }
 }
+
+export function getDaysEvents(day, events) {
+  if (!events || events.length === 0) {
+    return [];
+  }
+
+  const result = [];
+    
+  for (const event of events) {
+    const { date, date2 } = event,
+      date1 = date || event.date1;
+
+    if (!date1) {
+      continue;
+    }
+
+    if (date1.isSame(day.date) || (date2 && date1.isSameOrBefore(day.date) && date2.isSameOrAfter(day.date))) {
```

Comment:
Note: I'm going through all the events here for each day. Obviously we could improve this by keeping the events sorted (using something like https://www.npmjs.com/package/sorted-array, for example). For now, I'm just advancing this as a first pass, hoping to get the conversation going